packages/editor/src/browser/editor-navigation-contribution.ts

```diff
@@ -59,7 +60,14 @@ export class EditorNavigationContribution implements Disposable, FrontendApplica
         this.toDispose.pushAll([
             // TODO listen on file resource changes, if a file gets deleted, remove the corresponding navigation locations (if any).
             // This would require introducing the FS dependency in the editor extension.
-            this.editorManager.onCurrentEditorChanged(this.onCurrentEditorChanged.bind(this))
+            this.editorManager.onCurrentEditorChanged(this.onCurrentEditorChanged.bind(this)),
+            this.editorManager.onCreated(widget => {
+                this.locationStack.removeClosedEditor(widget.editor.uri);
+                widget.disposed.connect(() => this.locationStack.addClosedEditor({
+                    uri: widget.editor.uri,
+                    viewState: widget.editor.storeViewState()
+                }));
+            })
```

Comment:
```ts
this.editorManager.onCreated(widget => {
    const toDispose = editorWidget.onDidDispose(() => {
        this.locationStack.addClosedEditor({
            uri: widget.editor.uri,
            viewState: widget.editor.storeViewState()
        });
        toDispose.dispose();
    });
})
```
What do you think?

Reply:
thanks for the suggestion, but unfortunately, the `viewState` becomes null when I tried listening for `onDidDispose` from the editor widget.

Reply:
Then go with your initial proposal but you have to disconnect from the `disposed` listener, don't you?

Reply:
I looked into it, and it seems to me that we only have to disconnect if `disposed` is a listener. But in this case, `disposed` is an emitted signal, so we probably don't need to disconnect.

Reply:
Cool. Thanks for checking it 👍